sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -78,29 +78,66 @@ class SessionCatalog(
     if (conf.caseSensitiveAnalysis) name else name.toLowerCase
   }
 
+  /**
+   * Validate names
+   */
+  protected[this] def validateName(name: String): Boolean = {
+    val validName = "([_A-Za-z0-9]+)".r
```

Comment:
This compiles a regex every time the method gets invoked. Can you move this into an object or the class body?